faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -200,8 +199,7 @@ public Connection newSessionConnection(String authToken) {
    */
   @Override
   public void close() {
-    if (closed.compareAndSet(false, true))
```

Comment:
We can't allow multiple calls to `client.close()` as it decreases its `refCount`. I think the `closed` flag is still needed.

Reply:
isn't that a workaround? I mean, the user shouldn't be calling `close()` multiple times

Reply:
The user shouldn't call it multiple times but, if that happens, the side effect is to corrupt the ref count of the underneath http connection.